pkg/virtctl/imageupload/imageupload.go

```diff
@@ -394,6 +397,12 @@ func (c *command) run(args []string) error {
 		fmt.Printf("Uploading %s completed successfully\n", imagePath)
 	}
 
+	if dataSource {
```

Comment:
Do we need to reword L392 to avoid folks ctrl-c'ing early before the DataSource is created?

Reply:
You're right, my suggestion would be then to place DS creation right after these lines:

https://github.com/kubevirt/kubevirt/blob/4dad3440327c6a1a26cda6f706030bc06c850f01/pkg/virtctl/imageupload/imageupload.go#L328-L338

WDYT?

Reply:
What about moving it after `uploadData` and handling the case where a `DataSource` already exists with the same name? That way we always get the `DataSource` regardless of the user ctrl-c'ing the CLI *and* if they do rerun we just reuse the existing `DataSource`?

Reply:
Sounds good, will do!